rfc/rfc-61/rfc-61.md

```diff
@@ -0,0 +1,240 @@
+   
+## Implementation
+
+![basic_arch](basic_arch.png)
+
+### Extend Savepoint meta 
+Snapshot view need to extend the savepoint metadata, so we are going to add one struct with four fields: 
+* tag_name: tag name for your snapshot
+* retain-days: number of day, So data belongs to this snapshot will be retained for retain-days,  then can be clean after snapshot expire
+* database: database name in Catalog
+* table-name: table name in Catalog 
+
+new Savepoint Metadata should look like below:
+``` diff
+{
+	"type": "record",
+	"name": "HoodieSavepointMetadata",
+	"namespace": "org.apache.hudi.avro.model",
+	"fields": [{
+		"name": "savepointedBy",
+		"type": {
+			"type": "string",
+			"avro.java.string": "String"
+		}
+	}, {
+		"name": "savepointedAt",
+		"type": "long"
+	}, {
+		"name": "comments",
+		"type": {
+			"type": "string",
+			"avro.java.string": "String"
+		}
+	}, {
+		"name": "partitionMetadata",
+		"type": {
+			"type": "map",
+			"values": {
+				"type": "record",
+				"name": "HoodieSavepointPartitionMetadata",
+				"fields": [{
+					"name": "partitionPath",
+					"type": {
+						"type": "string",
+						"avro.java.string": "String"
+					}
+				}, {
+					"name": "savepointDataFile",
+					"type": {
+						"type": "array",
+						"items": {
+							"type": "string",
+							"avro.java.string": "String"
+						}
+					}
+				}]
+			},
+			"avro.java.string": "String"
+		}
+	}, {
+		"name": "version",
+		"type": ["int", "null"],
+		"default": 1
++	}, {
++		"name": "tag",
++		"type": "record",
++	    "fields": [{
++		        "name": "tag_name",
++			    "type": "string",
++		    	"avro.java.string": "String"
++		    }, 
++           {
++    		    "name": "retain_days",
++    	    	"type": ["int", "null"],
++    	    	"default": 0
++    	    },
++           {
++		        "name": "database",
++			    "type": "string",
++		    	"avro.java.string": "String"
++		    }, 
++           {
++		        "name": "table_name", // will infer tag_name's value if doesn't specific
++			    "type": "string",
++		    	"avro.java.string": "String"
++		    }, 
++	    ]
++     }
++	]
+}
+```
+
+### Meta Sync
+Create a snapshot view also will create a new external table into the Catalogs, and add a timestamp into tbl properties to identify which savepoint you are using.
+for example, if you choose Hive Metastore as the catalog and create a snapshot view on a Hudi table, following steps will be process:
+* create a new savepoint with tag name (let's say tbl-YYYYMMDD)
+* create an external table in HMS, table's name is tbl-YYYYMMDD
+* add as.of.instant= ${savepoint's timestamp} into table tbl-YYYYMMDD's storage properties ，
+
+when user query such a snapshot's external table, engines like Spark/Presto will get the savepoint timestamp from external table's properties then pass back to Hudi for time travel 
+
+### Clean service
+Normal savepoint will never be cleaned in Clean service, but a tagged savepoint is cleanable since it could be out-of-date.
```

Comment:
no need to distinguish normal vs tagged savepoint. just let cleaner check expiryTimestamp and then decide